### PR TITLE
Faster compute of burndown charts

### PR DESCRIPTION
**Pull request by eseliger**
Turns out for the burndown chart we really don't need all events in the PR timelines. Given that, we can add a filter by kind and sometimes drastically reduce the amount of events that need to be processed. In case of my 9k changesets campaign this reduces it to ~30% of the initial amount of events.

Before:
Required changesets: 9045, events: 106843, request duration: 5.34s

After:
Required changesets: 9045, events: 31752, request duration: 4.10s

### CODE REVIEW

File: enterprise/internal/campaigns/store_changeset_events_test.go [hunk 1]
```diff
@@ -44,7 +49,7 @@ func testStoreChangesetEvents(t *testing.T, ctx context.Context, s *Store, _ rep
 		}
 
 		// Verify that no duplicates are introduced and no error is returned.
-		for i := 0; i < 2; i++ {
+		for i := 0; i < len(events); i++ {
```
Comment (chrispine) [inline enterprise/internal/campaigns/store_changeset_events_test.go:52]: I know this is only a test, but this is a weird looking loop. Why is it being done `len(events)` number of times?

**Review by chrispine — APPROVED** [review]

File: enterprise/internal/campaigns/store_changeset_events_test.go (hunk 1 above)
Comment (eseliger) [reply enterprise/internal/campaigns/store_changeset_events_test.go:52]: whoups good catch. It's supposed to test the upsert function does actually upsert. Not needed to run more often 🤦 
```suggestion
		for i := 0; i < 2; i++ {
```

**Review by LawnGnome — APPROVED** [review]
Nice!

**Review by mrnugget — COMMENTED** [review]
I approve of every PR that includes "Faster compute" in its title. This one also backs up the claims 👍

**Pull request merged**